File: server_node_legacy/verify.js
const { exec } = require('child_process');

const BASE_URL = 'http://localhost:5000/api';

async function verify() {
    console.log('Starting verification with fetch...');

    try {
        // Helper for post request
        const post = async (url, body) => {
            const res = await fetch(BASE_URL + url, {
                method: 'POST',
                headers: { 'Content-Type': 'application/json' },
                body: JSON.stringify(body)
            });
            const data = await res.json();
            return { status: res.status, data };
        };

        // 1. Register
        console.log('1. Registering user...');
        const regRes = await post('/auth/register', {
            username: 'testu' + Date.now(),
            password: 'password123'
        });
        console.log('   Register status:', regRes.status);
        const token = regRes.data.token;

        // 2. Create Resource
        console.log('2. Creating Resource...');
        const resRes = await post('/resources', {
            name: 'Hall A',
            type: 'Room'
        });
        const resourceId = resRes.data.id;
        console.log('   Resource created:', resourceId);

        // 3. Create Events
        console.log('3. Creating Events...');
        const evt1 = await post('/events', {
            title: 'Workshop 1',
            start_time: '2023-10-10T10:00:00Z',
            end_time: '2023-10-10T12:00:00Z',
            description: 'First'
        });
        const evt2 = await post('/events', {
            title: 'Workshop 2',
            start_time: '2023-10-10T11:00:00Z', // Overlaps
            end_time: '2023-10-10T13:00:00Z',
            description: 'Second'
        });
        console.log(`   Events created: ${evt1.data.id}, ${evt2.data.id}`);

        // 4. Allocate Event 1 -> Success
        console.log('4. Allocating Workshop 1 to Hall A...');
        const alloc1 = await post('/allocations', {
            event_id: evt1.data.id,
            resource_id: resourceId
        });
        if (alloc1.status === 200) console.log('   Allocation successful.');
        else console.error('   Failed allocation:', alloc1.data);

        // 5. Allocate Event 2 -> Fail (Conflict)
        console.log('5. Allocating Workshop 2 to Hall A (expect conflict)...');
        const alloc2 = await post('/allocations', {
            event_id: evt2.data.id,
            resource_id: resourceId
        });

        if (alloc2.status === 409) {
            console.log('   ✅ SUCCESS: Conflict rejected correctly (409).');
        } else {
            console.error('   ❌ FAILED: Unexpected status:', alloc2.status, alloc2.data);
        }

    } catch (err) {
        console.error('Global Verification Error:', err.message);
    }
}

setTimeout(verify, 1000);

File: server/verify.py
import requests
import json
import time
from datetime import datetime

BASE_URL = 'http://localhost:5000/api'

def run_test():
    print("Starting Flask Verification...")
    
    # 1. Register
    print("1. Registering user...")
    try:
        res = requests.post(f"{BASE_URL}/auth/register", json={
            "username": f"testuser_{int(time.time())}",
            "password": "password123"
        })
        print(f"   Register Status: {res.status_code}")
        token = res.json().get('token')
    except Exception as e:
        print(f"   Error: {e}")
        return

    # 2. Create Resource
    print("2. Creating Resource...")
    res = requests.post(f"{BASE_URL}/resources", json={
        "name": "Meeting Room 1",
        "type": "Room"
    })
    resource_id = res.json().get('id')
    print(f"   Resource ID: {resource_id}")

    # 3. Create Events
    print("3. Creating Events...")
    # Event 1: 10:00 - 12:00
    res1 = requests.post(f"{BASE_URL}/events", json={
        "title": "Strategy Meeting",
        "start_time": "2023-10-25T10:00:00",
        "end_time": "2023-10-25T12:00:00",
        "description": "Important"
    })
    evt1_id = res1.json().get('id')

    # Event 2: 11:30 - 13:30 (Overlap)
    res2 = requests.post(f"{BASE_URL}/events", json={
        "title": "Lunch & Learn",
        "start_time": "2023-10-25T11:30:00",
        "end_time": "2023-10-25T13:30:00",
        "description": "Fun"
    })
    evt2_id = res2.json().get('id')
    print(f"   Events Created: {evt1_id}, {evt2_id}")

    # 4. Allocate 1 -> Success
    print("4. Allocating Event 1...")
    res = requests.post(f"{BASE_URL}/allocations", json={
        "event_id": evt1_id,
        "resource_id": resource_id
    })
    if res.status_code == 200:
        print("   [OK] Allocation 1 Successful")
    else:
        print(f"   [FAIL] Allocation 1 Failed: {res.text}")

    # 5. Allocate 2 -> Fail
    print("5. Allocating Event 2 (Conflict)...")
    res = requests.post(f"{BASE_URL}/allocations", json={
        "event_id": evt2_id,
        "resource_id": resource_id
    })
    if res.status_code == 409:
        print("   [OK] Conflict Detected Correctly (409)")
    else:
        print(f"   [FAIL] Failed to detect conflict. Status: {res.status_code}")

if __name__ == "__main__":
    time.sleep(2) # Give server a moment
    run_test()
